### ISSUE TITLE
`scipy` convergence but Jacobian still large

### ISSUE THREAD

**Issue by skhilton** [jbloomlab/phydms#14]
Here is a run with a large `H1` tree with the following error: 

```
RuntimeError: Even though optimizer reports convergence, the maximum element in the Jacobian is too large: 2903.287319203765
Here is the summary from the optimization:
['Step 1: optimized parameters, loglik went from -19873.64 to -17170.75 (18 iterations, 27 function evals)', 'Step 2: optimized branches, loglik went from -17170.75 to -17168.14 (50 iterations, 59 function evals)', 'Step 3: optimized parameters, loglik went from -17168.14 to -17168.12 (4 iterations, 8 function evals)', 'Step 4: optimized branches, loglik went from -17168.12 to -17168.12 (1 iterations, 4 function evals)']
[jacobian_error.zip](https://github.com/jbloomlab/phydms/files/1002039/jacobian_error.zip)

```

**Comment by jbloom** [comment]
@skhilton: It doesn't look like Markdown inclusions actually attach the file. So can you attach the `jacobian_error.zip` file manually to your issue using the `selecting them` link at the bottom of the GitHub issue text entry box.

**Comment by skhilton** [comment]
[jacobian_error.zip](https://github.com/jbloomlab/phydms/files/1002499/jacobian_error.zip)

**Comment by jbloom** [comment]
It should now be fixed in `2.0.5`. It turns out the Jacobian can sometimes have elements that are still fairly large for extremely short branch lengths, since these have a limit of 1e-6. Now the Jacobian check is only implemented for model parameter optimization. Hopefully this fixes all the issues!
